crates/nu-cmd-lang/src/example_support.rs

```diff
@@ -159,11 +160,19 @@ pub fn check_example_evaluates_to_expected_output(
     // If the command you are testing requires to compare another case, then
     // you need to define its equality in the Value struct
     if let Some(expected) = example.result.as_ref() {
-        assert_eq!(
-            DebuggableValue(&result),
-            DebuggableValue(expected),
-            "The example result differs from the expected value",
-        )
+        let expected = DebuggableValue(expected);
+        let result = DebuggableValue(&result);
+        if result != expected {
+            panic!(
+                "\x1b[31mError:\x1b[0m The result of example \x1b[34m'{}'\x1b[0m for the command \x1b[34m'{}'\x1b[0m differs from the expected value.\n\n\
+                Expected: {:?}\n\
+                Actual:   {:?}\n",
+                example.description,
+                cmd_name,
+                expected,
+                result
+            );
+        }
```

Comment:
Thanks for the addded formatting! But I think `assert_eq` also allows a format string:

```suggestion
        assert_eq!(
            result,
            expected,
            "\x1b[31mError:\x1b[0m The result of example \x1b[34m'{}'\x1b[0m for the command\x1b[34m'{}'\x1b[0m differs from the expected value.\n\n\
Expected: {:?}\n
Actual:   {:?}\n",
                example.description,
                cmd_name,
                expected,
                result,
            );
        }
```

Hopefully, that formatting looks right?

Reply:
For some reason it doesn't format the empty space the same as panic - so I have put it all on one line now